src/test/java/cryptator/ParserTest.java

```diff
@@ -121,5 +121,73 @@ public void testParserError7() throws CryptaParserException {
 		parser.parse("[send + more] >= money");
 	}
 
+	@Test
+	public void testParserAND() throws CryptaParserException {
+		final ICryptaNode node = parser.parse("send+more=money; d+e>=y");
+
+		testPreorder("; = + send more money >= + d e y ", node);
+		testPostorder("send more + money = d e + y >= ; ", node);
+		testInorder("send + more = money ; d + e >= y ", node);
+	}
+
+	@Test
+	public void testParserAND2() throws CryptaParserException {
+		final ICryptaNode node = parser.parse("send+more=	money; -send -more= \n -money");
+
+		testPreorder("; = + send more money = - - "+ ZERO+" send more - "+ ZERO+" money ", node);
+		testPostorder("send more + money = "+ ZERO+" send - more - "+ ZERO+" money - = ; ", node);
+		testInorder("send + more = money ; "+ ZERO+" - send - more = "+ ZERO+" - money ", node);
+
+	}
+
+	@Test
+	public void testParserAND3() throws CryptaParserException {
+		var str = "send + more ^ more = money ";
+		final ICryptaNode node = parser.parse(str+";"+str);
+		var preord = "= + send ^ more more money ";
+		testPreorder("; " + preord + preord, node);
+		var postord = "send more more ^ + money =";
+		testPostorder(postord + " " + postord + " ; ", node);
+		testInorder(str + "; " + str, node);
+	}
+
+	@Test(expected = CryptaParserException.class)
+	public void testParserErrorAND() throws CryptaParserException {
```

Comment:
Is it an error ? Or do we skip the empty equation ?

Reply:
Normally the grammar doesn't accept an empty equation. It is a conceptional choice, do you think it is better to allow it ? In this case, we will authorize for example the expression`;;;;;;;;`

Reply:
Most parsers accept empty instructions, for instance in Java. 
So, I would accept `send+more=money;` or `send+more=money;;;;`, but would refuse an empty cryptarithm `;;;;`.
It is more user friendly.

Reply:
Ok, I correct it as well

Reply:
Thanks

Reply:
Last question, if we accept `send+more=money;;;;` and want to print it for example in infixorder, then the redondant `;` will disappear in favour of a _light_ print of this form `send + more = money` ?

Same here : `send+more=money;;; d+e>=y` -> `send + more = money ; d + e >= y` ?